Tensile/KernelWriterAssembly.py

```diff
@@ -3325,17 +3325,88 @@ def getKernArg(self, parmName, writeSgpr=1):
   # code phrase for load batched address from array of buffer pointer
   ##############################################################################
   def loadBatchedAddress(self, kernel, Batch, tmpSgpr):
+    laneSC = self.laneSGPRCount
     kStr = self.endLine
 
-    for idx in kernel["ProblemType"]["IndicesBatch"]:
-      if not isPackedIndex(kernel,idx):
-        kStr += inst("s_mul_i32", sgpr(tmpSgpr), sgpr(Batch), 0x8, "offset of global buffer address")
-        if not kernel["_GlobalAccumulation"]:
+    # handle Batch C/D
+    if not kernel["_GlobalAccumulation"]:
+      for idx in kernel["ProblemType"]["IndicesBatch"]:
+        if not isPackedIndex(kernel,idx):
+          kStr += inst("s_mul_i32", sgpr(tmpSgpr), sgpr(Batch), 0x8, "offset of global buffer address")
           kStr += inst("s_load_dwordx2", sgpr("AddressD", 2), sgpr("AddressD",2), sgpr(tmpSgpr), "load global buffer D address")
+
+      endCheckLabel = self.getNamedLabel(f"label_skip_c_buffer_deref_{Batch}")
+      if kernel["ProblemType"]["ComputeDataType"].isDoubleComplex():
+        kStr += inst("v_cmp_eq_f64", sgpr(tmpSgpr, laneSC), sgpr("Beta", 2), 0.0, "Beta.real == 0.0 ?")
+        kStr += inst("v_cmp_eq_f64", self.vcc, sgpr("Beta", 2), 0.0, "Beta.real == 0.0 ?")
```

Comment:
why check beta.real two times?
Is this beta.imag?

Reply:
yes, this is a mistake.

Reply:
fix it in new commit.